bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/helper/image/WOIDUrlBuilderHelper.java

```diff
@@ -0,0 +1,196 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.helper.image;
+
+import com.adobe.cq.wcm.core.components.models.Image;
+import com.adobe.cq.wcm.spi.AssetDelivery;
+import com.day.cq.commons.DownloadResource;
+import com.day.cq.commons.ImageResource;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.sling.api.resource.ValueMap;
+
+import java.awt.*;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class WOIDUrlBuilderHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WOIDUrlBuilderHelper.class);
+
+    private static String COMMA = ",";
+    private static String WIDTH_PARAMETER = "width";
+    private static String QUALITY_PARAMETER = "quality";
+    private static String CROP_PARAMETER = "c";
+    private static String ROTATE_PARAMETER = "r";
+    private static String FLIP_PARAMETER = "flip";
+    private static String SIZE_PARAMETER = "sz";
+    private static String FORMAT_PARAMETER = "format";
+    private static String PATH_PARAMETER = "path";
+    private static String SEO_PARAMETER = "seoname";
+
+
+    public static String getSrcSet(@NotNull AssetDelivery assetDelivery, @NotNull Resource assetResource, @NotNull String imageName,
```

Comment:
`assetResource` seems to be a bit misleading, looking at how you call the method it seems it's `imageComponentResource` actually, isn't it? Unless there's a bug there